src/main/flight/mixer.c

```diff
@@ -743,27 +742,115 @@ STATIC_UNIT_TESTED void servoMixer(void)
+        int16_t rollPitchYawMixMin = 0;
+
+        // Find roll/pitch/yaw desired output
+        for (i = 0; i < motorCount; i++) {
+            rollPitchYawMix[i] =
+                axisPID[PITCH] * currentMixer[i].pitch +
+                axisPID[ROLL] * currentMixer[i].roll +
+                -mixerConfig->yaw_motor_direction * axisPID[YAW] * currentMixer[i].yaw;
+
+            if (rollPitchYawMix[i] > rollPitchYawMixMax) rollPitchYawMixMax = rollPitchYawMix[i];
+            if (rollPitchYawMix[i] < rollPitchYawMixMin) rollPitchYawMixMin = rollPitchYawMix[i];
+        }
+
+        // Scale roll/pitch/yaw uniformly to fit within throttle range
+        int16_t rollPitchYawMixRange = rollPitchYawMixMax - rollPitchYawMixMin;
+        int16_t throttleRange, throttle;
+        int16_t throttleMin, throttleMax;
+
+        // Find min and max throttle based on condition
+        if (feature(FEATURE_3D)) {
+            static int16_t throttlePrevious = 0;   // Store the last throttle direction for deadband transitions
+
+            if (rcData[THROTTLE] <= (rxConfig->midrc - flight3DConfig->deadband3d_throttle)) { // Out of deadband handling
+                throttleMax = flight3DConfig->deadband3d_low;
+                throttleMin = escAndServoConfig->minthrottle;
+                throttlePrevious = throttle = rcCommand[THROTTLE];
+            } else if (rcData[THROTTLE] >= (rxConfig->midrc + flight3DConfig->deadband3d_throttle)) {
+                throttleMax = escAndServoConfig->maxthrottle;
+                throttleMin = flight3DConfig->deadband3d_high;
+                throttlePrevious = throttle = rcCommand[THROTTLE];
+            } else {  // Deadband handling
+                // Always start positive. When coming from positive keep positive throttle within deadband
+                if ((throttlePrevious >= (rxConfig->midrc + flight3DConfig->deadband3d_throttle)) || !throttlePrevious) {
+                    throttleMax = escAndServoConfig->maxthrottle;
+                    throttle = throttleMin = flight3DConfig->deadband3d_high;
+                // When coming from negative. Keep negative throttle within deadband
+                } else if (throttlePrevious <= (rxConfig->midrc - flight3DConfig->deadband3d_throttle)) {
+                    throttle = throttleMax = flight3DConfig->deadband3d_low;
+                    throttleMin = escAndServoConfig->minthrottle;
+                }
+            }
+        } else {
+            throttle = rcCommand[THROTTLE];
+            throttleMin = escAndServoConfig->minthrottle;
+            throttleMax = escAndServoConfig->maxthrottle;
+        }
+
+        throttleRange = throttleMax - throttleMin;
+
+        if (rollPitchYawMixRange > throttleRange) {
+            motorLimitReached = true;
+            for (i = 0; i < motorCount; i++) {
+                rollPitchYawMix[i] = (rollPitchYawMix[i] * throttleRange) / rollPitchYawMixRange;
```

Comment:
@ledvinap I actually think that this line may cause overflow of int16_t. That may explain the "tasmanian devil effect" I had when centering the throttle.